internal/lang/eval.go

```diff
@@ -453,6 +453,12 @@ func (s *Scope) evalContext(refs []*addrs.Reference, selfAddr addrs.Referenceabl
 
 	if len(outputValues) > 0 {
 		vals["output"] = cty.ObjectVal(outputValues)
+
+		// "run" block will have output values when size of ref.Remaining is greater than 0.
+		if len(refs) > 0 && len(refs[0].Remaining) > 0 {
```

Comment:
I may not be fully understanding the code in this area, but what if `refs[0].Remaining` was 0 and `refs[1].Remaining` was > 0, shouldn't we also handle that?

I think we shouldn't just check the 0th element in the slice.